### ISSUE TITLE
Keep track of actual frame timing

### ISSUE THREAD

**Issue by kattjevfel** [matanui159/ReplaySorcery#15]
Not quite sure how to describe this, but when recording my desktop it's all good, but if I run a game and record, the recording is sped up and goes beyond the set time limit (encoded video is the proper length but actual content has passed more time)

```log
Jul 28 18:30:38 main replay-sorcery[117157]: Output progress: 1800/1800
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: frame I:8     Avg QP:23.75  size:422106
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: frame P:1792  Avg QP:27.16  size:104224
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: mb I  I16..4: 100.0%  0.0%  0.0%
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: mb P  I16..4:  8.3%  0.0%  0.0%  P16..4: 31.3%  0.0%  0.0%  0.0%  0.0%    skip:60.4%
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: coded y,uvDC,uvAC intra: 76.3% 47.4% 16.8% inter: 21.1% 11.6% 2.0%
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: i16 v,h,dc,p: 13% 28% 42% 17%
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: i8c dc,h,v,p: 40% 33% 17%  9%
Jul 28 18:30:38 main replay-sorcery[117157]: x264 [info]: kb/s:50705.87
Jul 28 18:30:39 main replay-sorcery[117157]: Running command: notify-send ReplaySorcery "Video saved!"
```

If I had to guess it's probably missing frames and the program just grabs a set amount of frames.

Here's a recording shown along with an analog clock: https://giant.gfycat.com/InfatuatedElasticFluke.mp4
While it looks like I'm a real epic rally driver I'd rather stick to the harsh truth here.

Running (at the moment) latest commit 215a9eef0cbbc9a5fdbaa9637cec7ea3101d6739
Other things that might be relevant: Arch, Linux 5.7.10-zen1-1-zen, Plasma 5.19.3, Ryzen 3700X, Vega 56, 16GB RAM

**Comment by matanui159** [comment]
This is due to it not doing proper frame pacing, so if there is lag in the recording or frames are dropped it doesnt keep track of the real timings and assumes they're still `1/framerate` seconds from each other

**Comment by matanui159** [comment]
Proper frame timing has been implemented by the FFmpeg rewrite (#37).
